app.js

```diff
@@ -9,19 +9,31 @@ const connection = require("./db/connection");
 //require controllers
 const getTopics = require("./controllers/topics.controller")
 const getEndpoints = require("./controllers/endpoints.controller")
+const getArticleById = require("./controllers/articles.controller")
 
-//ENDPOINTS
+////ENDPOINTS////
 app.get("/api/topics", getTopics);
 app.get("/api", getEndpoints);
+app.get("/api/articles/:article_id", getArticleById);
 
-//ERROR HANDLING
-//Bad path
+////ERROR HANDLING////
+//400 Bad Path
+//No err parameter as should be triggered where no endpoint reached
 app.use((req, res, next) => {
-    res.status(400).send({ msg: "Path does not exist."});
+    console.log("BAD PATH BABY")
+    return res.status(400).send({ msg: "Path does not exist."});
 })
-//"Catch all bucket"
-app.use("*", (req, res, next) => {
-    res.status(500).send({ msg: "Internal server error."})
+//404 Nothing found (error received from model)
+app.use((err, req, res, next) => {
+    if (err.status && err.msg) {
+        return res.send(err);
+    }
+    return res.status(404).send({ msg: "Not found."});
+})
```

Comment:
Good work here, I can see you're checking that there is a `.status` property, so maybe you could extend functionality and send `res.status(err.status)` instead of `res.status(404)` and make this reusable depending on what status and msg is given to it? These notes should be illuminate the point: https://notes.northcoders.com/courses/js-back-end/error-handling#custom-errors